### ISSUE TITLE
ElasticPress breaks Printful and Shipwire plugins (REST API / webhook related)

### ISSUE THREAD

**Issue by ahmed-sigmalux** [10up/ElasticPress#968]
Have you searched for similar issues before submitting this one?

Yes.

Is this a bug, question or feature request?

Bug.

Describe the issue you encountered:

ElasticPress is breaking two other WooCommerce plugins:

* Printful - https://www.printful.com/landing/woocommerce-print-products-fulfillment
* Shipwire - https://woocommerce.com/products/shipwire/

These plugins stop working whenever ElasticPress is enabled. With ElasticPress enabled, Printful returns the error:

```
While adding product to Your store we encountered connection timeout!

Check that Your server has enought system resources available to handle the request in less than 90 seconds, try pushing product with less variants and try increasing available system resources, if possible.

If the error persists feel free to contact our support by providing the code below.

CODE: G15179888
```

While Shipwire returns the error:

```
[Webhook Error] Invalid request method. 
Request method: POST 
uri: https://api.shipwire.com/api/v3/webhooks 
user-agent: WooCommerce-Shipwire/2.2.3 (WooCommerce/3.2.1; WordPress/4.8.3) 
headers: Array 
( [content-type] => application/json 
[accept] => application/json 
[Authorization] => ********************************************************************************** ) 
body: {"topic":"v1.order.updated","url":"https:\/\/mysite.com?wc-api=wc_shipwire_order_webhook"} 

Response code: 400 
message: Bad Request 
headers: Array 
( [server] => Apache 
[vary] => Accept-Encoding 
[content-type] => application/json 
[content-encoding] => gzip 
[p3p] => policyref="/w3c/p3p.xml", CP="COM CNT INT NAV PUR STA UNI CAO DSP CUR o i OUR IND" 
[strict-transport-security] => max-age=31536000; includeSubDomains 
[set-cookie] => sanitized
[content-length] => 254 ) 
body: {"status":400,"message":"Invalid request","errors":["Invalid URL"],"resourceLocation":"https:\/\/api.shipwire.com\/api\/v3\/webhooks?offset=0&limit=20","resource":{"offset":0,"total":1,"previous":null,"next":null,"items":[{"resourceLocation":null,"resource":{"id":0,"topic":"v1.order.updated","url":"https:\/\/mysite.com?wc-api=wc_shipwire_order_webhook"}}]}} 
```

Based on the above error messages, I'm led to believe ElasticPress is somehow interfering with the REST API or webhooks. I've enabled debug logging but it doesn't reveal anything useful.

Current WordPress version:

4.8.3

Current ElasticPress version:

2.4

Current Elasticsearch version:

5.6.3

Where do you host your Elasticsearch server:

Self-hosted

Other plugins installed (WooCommerce, Simple Redirect Manager, etc..):

WooCommerce, Yoast, Printful, and Shipwire

Steps to reproduce:

1. Start with a fresh WooCommerce and ElasticPress setup
2. Install Printful and / or Shipwire
3. Perform a sync for Printful / Shipwire and everything works
4. Install and enable ElasticPress and perform a manual index
5. Perform a sync for Printful / Shipwire and you get the above error messages

Screenshots, if needed:

**Comment by eugene-manuilov** [comment]
Hi @ahmed-sigmalux!

how many products do you have in your Printful account? I think it happens because ElasticPress tries to index products on the fly what reduces response time and lead to a situation when Printful considers sync request timed out.

Also could you please let me know how often you need to sync your products from Printful into WooCommerce? And can you sync by smaller chunks?

**Comment by ahmed-sigmalux** [comment]
The first time I saw that error, I thought the problem was trying to sync too many products at once. But I get the exact same error even when trying to sync a single product without any variations. The same goes for Shipwire - the plugin fails with just one product bring synced.

I spoke to Printful support and they said their server logs show that WooCommerce is accepting the REST POST but then not responding with an OK. When I disable ElasticPress, WooCommerce responds with an OK in just a few seconds.

I also spoke to SkyVerge (developer of the Shipwire plugin) and they said another plugin, which I've identified as ElasticPress, is causing the webhook endpoints to return invalid responses. Again, as soon as ElasticPress is disabled, the webhooks and the Shipwire plugin work fine.

For Printful, could ElasticPress on-the-fly indexing not be working or be taking more than the 90 seconds Printful's API allows for an OK response? For Shipwire, could ElasticPress be returning WooCommerce products in some non-standard format or manner that is causing the webhook to be malformed? Does ElasticPress have a debug or trace mode I can enable for better troubleshooting here?
